File: all-submissions/0202-happy-number/solution.java
class Solution {
    public boolean isHappy(int n) {
        int Square_Sum=n;
        while(Square_Sum!=1)
        {
             Square_Sum=sumsquare(Square_Sum);
            if(Square_Sum == 145)
                return false;
        }
        return true;
    }
    public int sumsquare(int n)
    {
        int sum = 0;
        while(n!=0)
        {
            sum += Math.pow(n%10,2);
            n /= 10;
        }
        return sum;
    }
}

File: all-submissions/0202-happy-number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        values = set()
        squares = [i**2 for i in range(10)]
        sum = 0
        while sum!=1:
            while(n != 0):
                unit = n%10
                sum += squares[unit]
                n = n //10
            if sum in values:
                break
            values.add(sum)
            n = sum
            sum = 0
        return True if sum == 1 else False

        
